Update hello world guide (#275)
solana-foundation/developer-content: content/guides/getstarted/hello-world-in-your-browser.md

---

For this "hello world" quickstart guide, we will use

> [Solana Playground](https://beta.solpg.io/https://github.com/solana-developers/hello_world_pg),

Reviewer: ```suggestion
[Solana Playground](https://beta.solpg.io/),
```

## Using Solana Playground

> [Solana Playground](https://beta.solpg.io/https://github.com/solana-developers/hello_world_pg)

Reviewer: ```suggestion
[Solana Playground](https://beta.solpg.io/)
```

is browser-based application that will let you write, build, and deploy onchain
Solana programs. All from your browser. No installation needed.

It is a great developer resource for getting started with Solana development,
especially on Windows.

### Import our example project

In a new tab in your browser, open our example "_Hello World_"

> [project on Solana Playground](https://beta.solpg.io/https://github.com/solana-developers/hello_world_pg)

Reviewer: Not sure which of these is the correct link.

solana airdrop 2
```

> Note: If you get a rate limit error when requesting devnet SOL you can also
> you can just type `run` in the terminal and you will get a link to a

> > [web faucet](https://faucet.solana.com/). You can also find your wallet
> > address at the bottom of the playground window.

Reviewer 2: markdown blockquotes are auto-converted to the blue "info" box. So writing "note" is now redundant

```suggestion
> If you get a rate limit error when requesting devnet SOL you can also you can
> just type `run` in the terminal and you will get a link to a
> [web faucet](https://faucet.solana.com/). You can also find your wallet
> address at the bottom of the playground window.
```

> [JSON RPC API](/docs/rpc.md) that reduced the need for rewriting common
> boilerplate, helping to simplify your client side application code.

### The JS client

> Now let's take a look at how to actually call our program. The example comes
> with a JS client that will call our hello world program. You can find the
> `client.ts` file on the left-hand side of the playground.

Reviewer 2: ```suggestion
Now let's take a look at how to actually call our program. The example comes
with a JavaScript client that will call our hello world program. You can find
the `client.ts` file on the left-hand side of the playground.
```

### Call the program

To execute your onchain program, you must send a

> [transaction](/docs/core/transactions.md) to the network. Each transaction
> submitted to the Solana blockchain contains a list of instructions.

Reviewer 2: lets go ahead of and link to instructions too so new people can read more

```suggestion
[transaction](/docs/core/transactions.md) to the network. Each transaction
submitted to the Solana blockchain contains a list of
[instructions](/docs/core/transactions.md#instruction).
```

### Run the application

Now that you know how the client works, you can run the code via the `run`
command. Just type it in the playground terminal and press enter.

```shell
run
```

Once your application completes, you will see output similar to this:

```shell
$ run
Running client...
  client.ts:
    ProgramID:  C7JcX81YDaDJ9Bf8ebifLgBSqfKJxETek6qyTuPGJE1f
    Transaction sent with hash: m7f7dszzdNshMZo5u2gRKjbyURk1tQHj7Hmeh3AbH7wUdnmDXmCJhA8cXJjYwVN7foJaLQiPYhEFTn6F5mWSeCb
    Congratulations! Look at your transaction in the Solana Explorer:
  https://explorer.solana.com/tx/m7f7dszzdNshMZo5u2gRKjbyURk1tQHj7Hmeh3AbH7wUdnmDXmCJhA8cXJjYwVN7foJaLQiPYhEFTn6F5mWSeCb?cluster=devnet
```

You can now click on the link provided to view your transaction in the Solana
Explorer. If you scroll to the bottom you should see `Hello, world!` in the

> **Log Messages** section of the program output. 🎉

Reviewer 2: why is "log messages" bold and capitalized?

Author: It was like that :D Maybe because it shows that its something in the playground terminal? Maybe nice? Any other suggestiong?

Reviewer 2: ahh that makes sense. we can just leave it then

#### Congratulations!!!

You have now written a Solana program, deployed it to devnet and called the
program. You are now a Solana developer!

PS: Try to update your program's message then re-build, re-deploy, and
re-execute your program.

### Look at the deployed program

Now that we called our program let's have a look at the account where the
program is actually saved.

> When executing a program using [web3.js](/docs/clients/javascript-reference.md)
> or from [another Solana program](/docs/core/cpi.md), you will need to provide

Reviewer: ```suggestion
When executing a program using [web3.js](/docs/clients/javascript-reference)
or from [another Solana program](/docs/core/cpi), you will need to provide
```

These work even if you add `.md`, so maybe you want to keep them, but the actual URLs don't have the `.md`

Author: Is there any difference between the link with .md or with?

Reviewer 2: having the .md extension in the file makes most code editors use auto complete. And will allow people to click on them when directly browing the markdown files on github.

I strongly recommend using them

Author: Okey, ill leave it as is then

Reviewer 2: I also just updated the contributing guide to state this too. I thought I had put it in there before, but I guess not

the `program id` (aka public address of your program).

Inside Solana Playground's **Build & Deploy** sidebar, you can find your
`program id` under the **Program Credentials** dropdown.

![Find Program Id](/assets/guides/hello-world-pg/pg-find-program-id.png)

> You can copy this program id and look at your program in the
> [solana explorer](https://explorer.solana.com/?cluster=devnet). Just paste in
> your program id in the search bar.

Reviewer 2: ```suggestion
You can copy this program id and look at your program in the
[Solana Explorer](https://explorer.solana.com/?cluster=devnet) by pasting the
address into the search bar.
```
